### ISSUE TITLE
Is it possible to access the old Language Reference pages?

### ISSUE THREAD

**Issue by per1234** [arduino/reference-en#209]
In the process of fixing regressions, I'm frequently wanting to look at the old reference pages so that I can port the changes here.

I did find that I could access the pre-change version of one page by using the previous URL:
https://www.arduino.cc/en/Reference/AttachInterrupt/

but with other pages that doesn't work because they redirect to the new page, for example:
https://www.arduino.cc/en/Reference/Constants
redirects to
https://www.arduino.cc/reference/en/language/variables/constants/constants

So, as a side note, I suspect that actually the lack of a redirect on https://www.arduino.cc/en/Reference/AttachInterrupt/ may be a bug.

**Comment by smellai** [comment]
You are right @per1234 , we are using this redirect map but some links are wrong:
```
/en/Reference/if	https://www.arduino.cc/reference/en/language/structure/control-structure/if/;
/en/Reference/else	https://www.arduino.cc/reference/en/language/structure/control-structure/else/;
/en/Reference/for	https://www.arduino.cc/reference/en/language/structure/control-structure/for/;
/en/Reference/switchCase	https://www.arduino.cc/reference/en/language/structure/control-structure/switchcase/;
/en/Reference/while	https://www.arduino.cc/reference/en/language/structure/control-structure/while/;
/en/Reference/doWhile	https://www.arduino.cc/reference/en/language/structure/control-structure/dowhile/;
/en/Reference/break	https://www.arduino.cc/reference/en/language/structure/control-structure/break/;
/en/Reference/continue	https://www.arduino.cc/reference/en/language/structure/control-structure/continue/;
/en/Reference/return	https://www.arduino.cc/reference/en/language/structure/control-structure/return/;
/en/Reference/goto	https://www.arduino.cc/reference/en/language/structure/control-structure/goto/;
/en/Reference/SemiColon	https://www.arduino.cc/reference/en/language/structure/further-syntax/semicolon/;
/en/Reference/Braces	https://www.arduino.cc/reference/en/language/structure/further-syntax/curlybraces/;
/en/Reference/Comments	https://www.arduino.cc/reference/en/language/structure/further-syntax/blockcomment/;
/en/Reference/Define	https://www.arduino.cc/reference/en/language/structure/further-syntax/define/;
/en/Reference/Include	https://www.arduino.cc/reference/en/language/structure/further-syntax/include/;
/en/Reference/Assignment	https://www.arduino.cc/reference/en/language/structure/arithmetic-operators/assignment/;
/en/Reference/Arithmetic	https://www.arduino.cc/reference/en/language/structure/arithmetic-operators/addition/;
/en/Reference/modulo	https://www.arduino.cc/reference/en/language/structure/arithmetic-operators/modulo/;
/en/Reference/Boolean	https://www.arduino.cc/reference/en/language/structure/boolean-operators/logicaland/;
/en/Reference/pointer	https://www.arduino.cc/reference/en/language/structure/pointer-access-operators/reference/;
/en/Reference/BitwiseAnd	https://www.arduino.cc/reference/en/language/structure/bitwise-operators/bitwiseand/;
/en/Reference/BitwiseXorNot	https://www.arduino.cc/reference/en/language/structure/bitwise-operators/bitwisenot/;
/en/Reference/Bitshift	https://www.arduino.cc/reference/en/language/structure/bitwise-operators/bitshiftleft/;
/en/Reference/increment	https://www.arduino.cc/reference/en/language/structure/compound-operators/increment/;
/en/Reference/incrementCompound	https://www.arduino.cc/reference/en/language/structure/compound-operators/compoundaddition/;
/en/Reference/bitwiseCompoundAnd	https://www.arduino.cc/reference/en/language/structure/compound-operators/compoundbitwiseand/;
/en/Reference/bitwiseCompoundOr	https://www.arduino.cc/reference/en/language/structure/compound-operators/compoundbitwiseor/;
/en/Reference/Constants	https://www.arduino.cc/reference/en/language/variables/constants/constants/;
/en/Reference/IntegerConstants	https://www.arduino.cc/reference/en/language/variables/constants/integerconstants/;
/en/Reference/Fpconstants	https://www.arduino.cc/reference/en/language/variables/constants/floatingpointconstants/;
/en/Reference/void	https://www.arduino.cc/reference/en/language/variables/data-types/void/;
/en/Reference/BooleanVariables	https://www.arduino.cc/reference/en/language/variables/data-types/boolean/;
/en/Reference/char	https://www.arduino.cc/reference/en/language/variables/data-types/char/;
/en/Reference/unsignedChar	https://www.arduino.cc/reference/en/language/variables/data-types/unsignedchar/;
/en/Reference/byte	https://www.arduino.cc/reference/en/language/variables/data-types/byte/;
/en/Reference/int	https://www.arduino.cc/reference/en/language/variables/data-types/int/;
/en/Reference/unsignedInt	https://www.arduino.cc/reference/en/language/variables/data-types/unsignedint/;
/en/Reference/word	https://www.arduino.cc/reference/en/language/variables/data-types/word/;
/en/Reference/long	https://www.arduino.cc/reference/en/language/variables/data-types/long/;
/en/Reference/unsignedLong	https://www.arduino.cc/reference/en/language/variables/data-types/unsignedlong/;
/en/Reference/short	https://www.arduino.cc/reference/en/language/variables/data-types/short/;
/en/Reference/float	https://www.arduino.cc/reference/en/language/variables/data-types/float/;
/en/Reference/double	https://www.arduino.cc/reference/en/language/variables/data-types/double/;
/en/Reference/string	https://www.arduino.cc/reference/en/language/variables/data-types/string/;
/en/Reference/StringObject	https://www.arduino.cc/reference/en/language/variables/data-types/stringobject/;
/en/Reference/array	https://www.arduino.cc/reference/en/language/variables/data-types/array/;
/en/Reference/CharCast	https://www.arduino.cc/reference/en/language/variables/conversion/charcast/;
/en/Reference/ByteCast	https://www.arduino.cc/reference/en/language/variables/conversion/bytecast/;
/en/Reference/IntCast	https://www.arduino.cc/reference/en/language/variables/conversion/intcast/;
/en/Reference/WordCast	https://www.arduino.cc/reference/en/language/variables/conversion/wordcast/;
/en/Reference/LongCast	https://www.arduino.cc/reference/en/language/variables/conversion/longcast/;
/en/Reference/FloatCast	https://www.arduino.cc/reference/en/language/variables/conversion/floatcast/;
/en/Reference/Scope	https://www.arduino.cc/reference/en/language/variables/variable-scope--qualifiers/scope/;
/en/Reference/static	https://www.arduino.cc/reference/en/language/variables/variable-scope--qualifiers/static/;
/en/Reference/volatile	https://www.arduino.cc/reference/en/language/variables/variable-scope--qualifiers/volatile/;
/en/Reference/const	https://www.arduino.cc/reference/en/language/variables/variable-scope--qualifiers/const/;
/en/Reference/PROGMEM	https://www.arduino.cc/reference/en/language/variables/utilities/progmem/;
/en/Reference/sizeof	https://www.arduino.cc/reference/en/language/variables/utilities/sizeof/;
/en/Reference/Setup	https://www.arduino.cc/reference/en/language/functions/main/setup/;
/en/Reference/Loop	https://www.arduino.cc/reference/en/language/functions/main/loop/;
/en/Reference/pinMode	https://www.arduino.cc/reference/en/language/functions/digital-io/pinmode/;
/en/Reference/digitalWrite	https://www.arduino.cc/reference/en/language/functions/digital-io/digitalwrite/;
/en/Reference/digitalRead	https://www.arduino.cc/reference/en/language/functions/digital-io/digitalread/;
/en/Reference/analogReference	https://www.arduino.cc/reference/en/language/functions/analog-io/analogreference/;
/en/Reference/analogRead	https://www.arduino.cc/reference/en/language/functions/analog-io/analogread/;
/en/Reference/analogWrite	https://www.arduino.cc/reference/en/language/functions/analog-io/analogwrite/;
/en/Reference/analogReadResolution	https://www.arduino.cc/reference/en/language/functions/arduino-due-only/analogreadresolution/;
/en/Reference/analogWriteResolution	https://www.arduino.cc/reference/en/language/functions/arduino-due-only/analogwriteresolution/;
/en/Reference/tone	https://www.arduino.cc/reference/en/language/functions/advanced-io/tone/;
/en/Reference/noTone	https://www.arduino.cc/reference/en/language/functions/advanced-io/notone/;
/en/Reference/shiftOut	https://www.arduino.cc/reference/en/language/functions/advanced-io/shiftout/;
/en/Reference/shiftIn	https://www.arduino.cc/reference/en/language/functions/advanced-io/shiftin/;
/en/Reference/pulseIn	https://www.arduino.cc/reference/en/language/functions/advanced-io/pulsein/;
/en/Reference/millis	https://www.arduino.cc/reference/en/language/functions/time/millis/;
/en/Reference/micros	https://www.arduino.cc/reference/en/language/functions/time/micros/;
/en/Reference/delay	https://www.arduino.cc/reference/en/language/functions/time/delay/;
/en/Reference/delayMicroseconds	https://www.arduino.cc/reference/en/language/functions/time/delaymicroseconds/;
/en/Reference/min	https://www.arduino.cc/reference/en/language/functions/math/min/;
/en/Reference/max	https://www.arduino.cc/reference/en/language/functions/math/max/;
/en/Reference/abs	https://www.arduino.cc/reference/en/language/functions/math/abs/;
/en/Reference/constrain	https://www.arduino.cc/reference/en/language/functions/math/constrain/;
/en/Reference/map	https://www.arduino.cc/reference/en/language/functions/math/map/;
/en/Reference/pow	https://www.arduino.cc/reference/en/language/functions/math/pow/;
/en/Reference/sqrt	https://www.arduino.cc/reference/en/language/functions/math/sqrt/;
/en/Reference/sin	https://www.arduino.cc/reference/en/language/functions/trigonometry/sin/;
/en/Reference/cos	https://www.arduino.cc/reference/en/language/functions/trigonometry/cos/;
/en/Reference/tan	https://www.arduino.cc/reference/en/language/functions/trigonometry/tan/;
/en/Reference/CharacterAnalysis	https://www.arduino.cc/reference/en/;
/en/Reference/randomSeed	https://www.arduino.cc/reference/en/language/functions/random-numbers/randomseed/;
/en/Reference/random	https://www.arduino.cc/reference/en/language/functions/random-numbers/random/;
/en/Reference/lowByte	https://www.arduino.cc/reference/en/language/functions/bits-and-bytes/lowbyte/;
/en/Reference/highByte	https://www.arduino.cc/reference/en/language/functions/bits-and-bytes/highbyte/;
/en/Reference/bitRead	https://www.arduino.cc/reference/en/language/functions/bits-and-bytes/bitread/;
/en/Reference/bitWrite	https://www.arduino.cc/reference/en/language/functions/bits-and-bytes/bitwrite/;
/en/Reference/bitSet	https://www.arduino.cc/reference/en/language/functions/bits-and-bytes/bitset/;
/en/Reference/bitClear	https://www.arduino.cc/reference/en/language/functions/bits-and-bytes/bitclear/;
/en/Reference/bit	https://www.arduino.cc/reference/en/language/functions/bits-and-bytes/bit/;
/en/Reference/attachInterrupt	https://www.arduino.cc/reference/en/language/functions/external-interrupts/attachinterrupt/;
/en/Reference/detachInterrupt	https://www.arduino.cc/reference/en/language/functions/external-interrupts/detachinterrupt/;
/en/Reference/interrupts	https://www.arduino.cc/reference/en/language/functions/interrupts/interrupts/;
/en/Reference/noInterrupts	https://www.arduino.cc/reference/en/language/functions/external-interrupts/detachinterrupt/;
/en/Reference/Serial	https://www.arduino.cc/reference/en/language/functions/communication/serial/;
/en/Reference/Stream	https://www.arduino.cc/reference/en/language/functions/communication/stream/;
/en/Reference/MouseKeyboard	https://www.arduino.cc/reference/en/language/functions/usb-leonardo-and-due-only/keyboard/;
/en/Reference/StringConstructor	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/string/;
/en/Reference/StringCharAt	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/charat;
/en/Reference/StringCompareTo	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/compareto;
/en/Reference/StringConcat	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/concat;
/en/Reference/CStr	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/c_str;
/en/Reference/StringEndsWith	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/endswith;
/en/Reference/StringEquals	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/equals;
/en/Reference/StringEqualsIgnoreCase	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/equalsignorecase;
/en/Reference/StringGetBytes	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/getbytes;
/en/Reference/StringIndexOf	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/indexof;
/en/Reference/StringLastIndexOf	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/lastindexof;
/en/Reference/StringLength	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/length;
/en/Reference/StringRemove	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/remove;
/en/Reference/StringReplace	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/replace;
/en/Reference/StringReserve	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/reserve;
/en/Reference/StringSetCharAt	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/setcharat;
/en/Reference/StringStartsWith	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/startswith;
/en/Reference/StringSubstring	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/substring;
/en/Reference/StringToCharArray	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/tochararray;
/en/Reference/StringToInt	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/toint;
/en/Reference/StringToFloat	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/tofloat;
/en/Reference/StringToLowerCase	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/tolowercase;
/en/Reference/StringToUpperCase	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/touppercase;
/en/Reference/StringTrim	https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/trim;
/en/Reference/StringBrackets	https://www.arduino.cc/reference/en/language/variables/data-types/string/operators/elementaccess;
/en/Reference/StringPlus	https://www.arduino.cc/reference/en/language/variables/data-types/string/operators/concatenation;
/en/Reference/StringAppend	https://www.arduino.cc/reference/en/language/variables/data-types/string/operators/append;
/en/Reference/StringComparison	https://www.arduino.cc/reference/en/language/variables/data-types/string/operators/comparison;
/en/Reference/StringGT	https://www.arduino.cc/reference/en/language/variables/data-types/string/operators/greaterthan;
/en/Reference/StringGE	https://www.arduino.cc/reference/en/language/variables/data-types/string/operators/greaterthanorequalto;
/en/Reference/StringLT	https://www.arduino.cc/reference/en/language/variables/data-types/string/operators/lessthan;
/en/Reference/StringLE	https://www.arduino.cc/reference/en/language/variables/data-types/string/operators/lessthanorequalto;
/en/Reference/StringDF	https://www.arduino.cc/reference/en/language/variables/data-types/string/operators/differentfrom;
```
I'm going to remove the redirects to help you and the community with the review. Thank you!

**Comment by per1234** [comment]
Thanks, that made the work immensely easier.

I believe I've now mined the closed issues labeled `Component: Documentation` in the arduino/Arduino issue tracker fairly thoroughly. There may still be a few left in the really old issues that were not fixed until recently but the yield was getting pretty low so I stopped at June, 2014.

I've submitted PRs for all of the major regressions I found through that process except this one:
https://github.com/arduino/Arduino/issues/2496#issuecomment-66872901
(the comment by Paul Stoffregen)
I don't have the energy to tackle that right now so if someone else wants to take over you're welcome to it, otherwise I'll try to get to it soon.

I believe the rest of the work on my to-do list can be done without access to the previous pages but I suspect there are other regressions that were not exposed by looking through the closed issues. It's frustrating to see improvements to the documentation make over the last couple of years gone. I am really excited about this new reference system but it's unfortunate that it's all based on significantly outdated content. I'm guessing you'll want to get those redirects turned back on ASAP but it would be really nice it there was still some way for people working on the documentation content to still have access to the previous pages.
